tests/cli/test_rasa_export.py

```diff
@@ -0,0 +1,240 @@
+
+
+def test_prepare_pika_event_broker():
+    # mock a pika event broker
+    pika_broker = Mock(spec=PikaEventBroker)
+
+    # patch the spinner so we can execute the `_prepare_pika_producer()` function
+    pika_broker.is_ready.return_value = True
+
+    # noinspection PyProtectedMember
+    export._prepare_event_broker(pika_broker)
+
+    # the attributes are set as expected
+    assert not pika_broker.should_keep_unpublished_messages
+    assert pika_broker.raise_on_failure
+
+
+@pytest.mark.parametrize(
+    "current_timestamp,maximum_timestamp,endpoints_path,requested_ids,expected",
+    [
+        (1.0, None, None, None, "--minimum-timestamp 1.0"),
+        (1.0, None, None, ["5", "6"], "--minimum-timestamp 1.0 --conversation-ids 5,6"),
+        (1.0, 3.4, None, None, "--minimum-timestamp 1.0 --maximum-timestamp 3.4"),
+        (
+            1.0,
+            2.5,
+            "a.yml",
+            None,
+            "--endpoints a.yml --minimum-timestamp 1.0 --maximum-timestamp 2.5",
+        ),
+        (
+            1.0,
+            2.5,
+            "a.yml",
+            ["1", "2", "3"],
+            (
+                "--endpoints a.yml --minimum-timestamp 1.0 --maximum-timestamp 2.5 "
+                "--conversation-ids 1,2,3"
+            ),
+        ),
+    ],
+)
+def test_get_continuation_command(
+    current_timestamp: float,
+    maximum_timestamp: Optional[float],
+    endpoints_path: Optional[Text],
+    requested_ids: Optional[List[Text]],
+    expected: Text,
+):
+    migrator = MockMigrator()
+    migrator.maximum_timestamp = maximum_timestamp
+    migrator.endpoints_path = endpoints_path
+    migrator.requested_conversation_ids = requested_ids
+
+    # noinspection PyProtectedMember
+    assert (
+        export._get_continuation_command(migrator, current_timestamp)
+        == f"rasa export {expected}"
+    )
+
+
+def _add_conversation_id_to_event(event: Dict, conversation_id: Text):
+    event["sender_id"] = conversation_id
+
+
+# noinspection PyProtectedMember
+def test_export_trackers(tmp_path: Path, monkeypatch: MonkeyPatch):
+    endpoints_path = write_endpoint_config_to_yaml(
+        tmp_path, {"event_broker": {"type": "pika"}, "tracker_store": {"type": "sql"}}
+    )
+
+    # export these conversation IDs
+    all_conversation_ids = ["id-1", "id-2", "id-3"]
+
+    requested_conversation_ids = ["id-1", "id-2"]
+
+    # create namespace with a set of cmdline arguments
+    namespace = argparse.Namespace(
+        endpoints=endpoints_path,
+        conversation_ids=",".join(requested_conversation_ids),
+        minimum_timestamp=1.0,
+        maximum_timestamp=10.0,
+    )
+
+    # prepare events from different senders and different timestamps
+    event_1 = random_user_uttered_event(1)
+    event_2 = random_user_uttered_event(2)
+    event_3 = random_user_uttered_event(3)
+    event_4 = random_user_uttered_event(4)
+    event_5 = random_user_uttered_event(11)
+    event_6 = random_user_uttered_event(5)
```

Comment:
list comprehension?

Reply:
that would be inconvenient as the events are used later on in the test - accessing them via their list index is more difficult to read